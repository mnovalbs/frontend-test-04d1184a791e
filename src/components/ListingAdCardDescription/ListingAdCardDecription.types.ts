export interface ListingAdCardDecriptionProps {
  description: string;
}
