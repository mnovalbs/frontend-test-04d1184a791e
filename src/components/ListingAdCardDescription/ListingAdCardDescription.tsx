import PhoneNumberMasker from "components/PhoneNumberMasker";
import { PHONE_NUMBER_REGEX } from "contants/regex";
import { splitPhoneNumber } from "utils/string";

import { ListingAdCardDecriptionProps } from "./ListingAdCardDecription.types";
import styles from "./ListingAdCardDecription.module.css"
import { Fragment } from "react";

function ListingAdCardDecription({ description }: ListingAdCardDecriptionProps) {
  const splittedDescription = splitPhoneNumber(description);

  return (
    <div className={styles.description} data-testid="description">
      {splittedDescription.map((descriptionChunk, index) => {
        if (PHONE_NUMBER_REGEX.test(descriptionChunk)) {
          return <PhoneNumberMasker key={index} phoneNumber={descriptionChunk} />;
        }
        return <Fragment key={index}>{descriptionChunk}</Fragment>;
      })}
    </div>
  );
}

export default ListingAdCardDecription;
