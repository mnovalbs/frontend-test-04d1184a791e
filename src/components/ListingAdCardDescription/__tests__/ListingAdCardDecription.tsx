import { render } from "@testing-library/react"
import mockAd from "contants/mockAd"
import ListingAdCardDescription from "../ListingAdCardDescription"

describe('ListingAdCardDescription', () => {
  it('renders description', () => {
    const { container } = render(<ListingAdCardDescription description={mockAd.description} />)

    expect(container).toMatchSnapshot()
  })
})