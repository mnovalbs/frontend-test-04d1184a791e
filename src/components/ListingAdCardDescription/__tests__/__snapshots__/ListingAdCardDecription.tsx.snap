// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingAdCardDescription renders description 1`] = `
<div>
  <div
    class="description"
    data-testid="description"
  >
    Phone number: 
    <span
      class="phoneNumber"
    >
      8234 XXXX
    </span>
    

Superb 3 bedroom in sentosa for a very affordable price and a very nearby bus
station.

Please call Alex B (CEA No: R0123456) @  
    <span
      class="phoneNumber"
    >
      8234XXXX
    </span>
     or the owner (
    <span
      class="phoneNumber"
    >
      8345XXXX
    </span>
    ) to arrange for a viewing.
  </div>
</div>
`;
